modules/metastorage/src/test/java/org/apache/ignite/internal/metastorage/impl/MetaStorageDeployWatchesCorrectnessTest.java

```diff
@@ -0,0 +1,115 @@
+import org.apache.ignite.internal.metastorage.server.SimpleInMemoryKeyValueStorage;
+import org.apache.ignite.internal.raft.RaftManager;
+import org.apache.ignite.internal.testframework.IgniteAbstractTest;
+import org.apache.ignite.internal.vault.VaultManager;
+import org.apache.ignite.internal.vault.inmemory.InMemoryVaultService;
+import org.apache.ignite.network.ClusterService;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * Tests that check correctness of an invocation {@link MetaStorageManager#deployWatches()}.
+ */
+public class MetaStorageDeployWatchesCorrectnessTest extends IgniteAbstractTest {
+
+    /** Vault manager. */
+    private static VaultManager vaultManager;
+
+    @BeforeAll
+    public static void init() {
+        vaultManager = new VaultManager(new InMemoryVaultService());
+
+        vaultManager.start();
+    }
+
+    @AfterAll
+    public static void deInit() {
+        vaultManager.stop();
+    }
+
+    /**
+     * Returns a stream with test arguments.
+     *
+     * @return Stream of different types of Meta storages to to check.
+     * @throws Exception If failed.
+     */
+    private static Stream<MetaStorageManager> metaStorageProvider() throws Exception {
+        HybridClock clock = new HybridClockImpl();
+        String mcNodeName = "mc-node-1";
+
+        ClusterManagementGroupManager cmgManager = mock(ClusterManagementGroupManager.class);
+        ClusterService clusterService = mock(ClusterService.class);
+        LogicalTopologyService logicalTopologyService = mock(LogicalTopologyService.class);
+        RaftManager raftManager = mock(RaftManager.class);
+
+        when(cmgManager.metaStorageNodes()).thenReturn(completedFuture(Set.of(mcNodeName)));
+        when(clusterService.nodeName()).thenReturn(mcNodeName);
+        when(raftManager.startRaftGroupNodeAndWaitNodeReadyFuture(any(), any(), any(), any(), any())).thenReturn(completedFuture(null));
+
+        return Stream.of(
+                new MetaStorageManagerImpl(
+                        vaultManager,
+                        clusterService,
+                        cmgManager,
+                        logicalTopologyService,
+                        raftManager,
+                        new SimpleInMemoryKeyValueStorage(mcNodeName),
+                        clock
+                ),
+                StandaloneMetaStorageManager.create(vaultManager)
+        );
+    }
+
+    /**
+     * Invokes {@link MetaStorageManager#deployWatches()} and checks result.
+     *
+     * @param metastore Meta storage.
+     */
+    @ParameterizedTest
+    @MethodSource("metaStorageProvider")
+    public void testCheckCorrectness(MetaStorageManager metastore) {
+        var deployWatchesFut = metastore.deployWatches();
+
+        assertFalse(deployWatchesFut.isDone());
+
+        metastore.start();
```

Comment:
Same as for vault, MetaStorageManager is an IgniteCompoment thus it's required not only to start it but also call beforeNodeStop and stop. All in all, I'd suggest adding vault and ms to some components list and close it (beforeNodeStop + stop). From my point of view, it also may worth to change @Before**All**/@After**All** to Before**Each**/After**Each**.

Reply:
I added the methods' invocation for MC. But I do not think that have to be implemented in suggested manner.

Reply:
No, I add this in the test body.

Reply:
Not sure that I got you. I will add, or you've already added? I'm asking because currently I can see only vault's stop.